### ISSUE TITLE
H256 SRT gets blocked from time to time

### ISSUE THREAD

**Issue by timokorkalainen** [ossrs/srs#3060]
**Description(描述)**

If an SRT gets cut without a proper end of stream message from the sender, the stream gets stuck and the same stream id can't be reused until SRS is rebooted. This _may_ have something to do with H256 content and RTMP not supporting it.

Is there a way to prevent SRT to RTMP, as this may affect the situation? I use SRT again to pull the stream down to OBS.

1. SRS Version(版本): all released 4.0 versions so far

**Expect(期望行为)**

I expect the stream to be freed / ended with a preferably configurable timeout and to be reusable after that. I have not found such a timeout.

**Comment by xiaozhihong** [comment]
1. Prevent SRT to RTMP, can see this PR  #3010, using config srt_to_rtmp off;
2. Maybe you can use [peeridletimeout](https://github.com/ossrs/srs/wiki/v5_EN_SRTParams#peer_idle_timeout) to config the timeout of the stream.

**Comment by timokorkalainen** [comment]
Super, thanks, I will try these out and report back.

**Comment by timokorkalainen** [comment]
Ah, so none of those options are available yet in any Docker release? I tried to build the StateThread branch, but couldn't get it to work. I suppose I'll wait then for the official Docker versions. Good to know that is progressing.

**Comment by xiaozhihong** [comment]
> Ah, so none of those options are available yet in any Docker release? I tried to build the StateThread branch, but couldn't get it to work. I suppose I'll wait then for the official Docker versions. Good to know that is progressing.

Yes, we will be processing this PR in these two weeks, you can keep watch on it.

**Comment by timokorkalainen** [comment]
I got it to compile with no problems with SRT enabled in the config, I wasn't able to stream SRT in. Perhaps the config has changed in some way I didn't spot or something, or I missed some other parameter somewhere. Anyway, I can wait for the docker version. Thanks for the help so far!

**Comment by xiaozhihong** [comment]
> I got it to compile with no problems with SRT enabled in the config, I wasn't able to stream SRT in. Perhaps the config has changed in some way I didn't spot or something, or I missed some other parameter somewhere. Anyway, I can wait for the docker version. Thanks for the help so far!

Have any logs about errors or warnings? Now #3010 is merged into develop branch, docker version arriving soon.

**Comment by timokorkalainen** [comment]
I tried the new docker and with the same config I have working fine with v4, I am now getting the following error:
```
srt serve error code=6006 : srt disabled, vhost=
thread [1][67823h87]: do_cycle() [src/app/srs_app_srt_conn.cpp:260][errno=11](Resource temporarily unavailable)
```

**Comment by timokorkalainen** [comment]
I took the new 5.0.30 docker and can now run srt_to_rtmp off and the "common live" config from the wiki. I'll see if that changes the issue.

Did I understand correctly that peer_idle_timeout is not yet included in this docker version? All blocks I tried to include it in reported an illegal argument.

I suggest adding in the docs that you now need to include the srt {enabled on;} block in the vhost config. :)
